obs2ioda-v2/src/hsd.f90

```diff
@@ -302,85 +308,44 @@ subroutine read_HSD(ccyymmddhhnn, inpdir)
    do isegm = 1, nsegm
       ifile = isegm + (iband-1) * nsegm
       if ( .not. fexist(ifile) ) cycle
-      open(iunit, file=trim(fnames(ifile)), form='unformatted', access='stream', status='old')
-
-print*,'Reading from ', trim(fnames(ifile))
-      read(iunit) header%basic, &
-                  header%data,  &
-                  header%proj,  &
-                  header%navi,  &
-                  header%calib, &
-                  header%interCalib, &
-                  header%segm,  &
-                  header%navicorr%headerNum, &
-             header%navicorr%blockLen, &
-             header%navicorr%RoCenterColumn, &
-             header%navicorr%RoCenterLine, &
-             header%navicorr%RoCorrection, &
-             header%navicorr%correctNum
+      open(newunit=iunit, file=trim(fnames(ifile)), form='unformatted', action='read', access='stream', status='old', convert='little_endian')
+      print*,'Reading from ', trim(fnames(ifile))
+
+      read(iunit) header%basic%headerNum, header%basic%blockLen, header%basic%numHeader, header%basic%byteOrder, header%basic%satName, header%basic%procCenter, header%basic%obsArea, header%basic%dummy2, header%basic%hhnn, header%basic%obsStartTime, header%basic%obsEndTime, header%basic%fileCreateTime, header%basic%totalHeaderLen, header%basic%dataLen, header%basic%qcflag1, header%basic%qcflag2, header%basic%qcflag3, header%basic%qcflag4, header%basic%version, header%basic%fileName, header%basic%dummy40
```

Comment:
Do you really need to read one by one like this instead of reading just 'header%basic'? If you have to read this way, better use '&' to make this one line into multiple lines for better code readability.

Reply:
I am not sure why it does not work properly when reading using 'header%basic'. At this moment the only way I found that reads the data correctly is reading one by one.